operators/constellation-node-operator/controllers/nodeversion_controller.go

```diff
@@ -614,6 +614,17 @@ func (r *NodeVersionReconciler) createNewNodes(ctx context.Context, config newNo
 	if config.newNodesBudget < 1 || len(config.outdatedNodes) == 0 {
 		return nil
 	}
+	hasOutdatedControlPlanes := func() bool {
+		// We need to look at both the outdated nodes *and* the nodes that have already
+		// been moved to the donors here because even if a CP node has already been moved to
+		// the donors, we still want to defer worker upgrades until the new CP node is actually joined.
+		for _, entry := range append(config.outdatedNodes, config.donors...) {
+			if nodeutil.IsControlPlaneNode(&entry) {
+				return true
+			}
+		}
+		return false
+	}()
```

Comment:
```suggestion
	// We need to look at both the outdated nodes *and* the nodes that have already
	// been moved to the donors here because even if a CP node has already been moved to
	// the donors, we still want to defer worker upgrades until the new CP node is actually joined.
	hasOutdatedControlPlanes := false
	for _, entry := range append(config.outdatedNodes, config.donors...) {
		if nodeutil.IsControlPlaneNode(&entry) {
			hasOutdatedControlPlanes = true
		}
	}
```
Why inline this function if its not reused?